src/vinyldns.js

```diff
@@ -50,27 +50,55 @@ class VinylDNS {
   }
 
   deleteZone(id) {
-    return this._getOrDelete('zone', id, 'delete');
+    return this._getOrDelete(this.urls.zone(id), 'delete');
+  }
+
+  getRecordSets(zoneId, queryOpts) {
+    return this._getOrDelete(this.urls.getRecordSets(zoneId, queryOpts), 'get');
+  }
+
+  getRecordSet(details) {
+    return this._getOrDelete(this.urls.recordSet(details), 'get');
+  }
+
+  createRecordSet(recordSet) {
+    return this._createOrUpdate(recordSet, this.urls.recordSetsBase(recordSet.zoneId), 'post');
+  }
+
+  updateRecordSet(recordSet) {
```

Comment:
The `PUT` endpoints for the VinylDNS API unfortunately requires a URL segment (see: https://github.com/vinyldns/vinyldns/issues/406), so `this.urls.recordSet` will have to be used instead of `this.urls.recordSetsBase` here.

Testing locally, I used:
```
  updateRecordSet(recordSet, details) {
    return this._createOrUpdate(recordSet, this.urls.recordSet(details), 'put');
  }
```